safe-global/safe-wallet-web
# feat: recovery module removal

## What it solves

Resolves #2762

## How this PR fixes it

This adds a new recovery module removal flow, guarded by a confirmation modal as per [the designs](https://www.figma.com/file/pO43pjVwR34Uy1ihMsY3OM/Generic-recovery?type=design&node-id=1113-27149&mode=design&t=CU0dL7WaudsbDVBt-0). The flow/modal is also shown if removing a Delay Modifier via the module settings.

## How to test it

With recovery enabled, click the remove button from either the "Recovery" or "Modules" section of the settings. Observe the new dialog that, after confirming, allows for successful removal of the recovery module.

## Screenshots

![remove-recovery](https://github.com/safe-global/safe-wallet-web/assets/20442784/a6a5fe28-6565-4a89-ae34-7829e0cdedac)

## Checklist
